packages/polymath-issuer/src/actions/compliance.js

```diff
@@ -59,6 +86,214 @@ export type InvestorCSVRow = [
+            ['Removing Whitelist Manager'],
+            async () => {
+              const st: SecurityToken = getState().token.token.contract;
+              const permissionManager = await st.getPermissionManager();
+              if (permissionManager) {
+                const transferManager = await st.getTransferManager();
+                if (transferManager) {
+                  await permissionManager.changePermission(
+                    delegate,
+                    transferManager.address,
+                    PERM_TYPES.ADMIN,
+                    false
+                  );
+                }
+              }
+            },
+            'Whitelist Manager Removed',
+            () => {
+              dispatch(removeManager(delegate));
+            },
+            undefined,
+            undefined,
+            undefined,
+            true
+          )
+        );
+      },
+      `Remove the Investor Wallet from the Whitelist Managers List?`,
+      undefined,
+      'pui-large-confirm-modal'
+    )
+  );
+};
+
+export const archiveGeneralPermissionModule = () => async (
+  dispatch: Function,
+  getState: GetState
+) => {
+  const st: SecurityToken = getState().token.token.contract;
+  dispatch(
+    ui.tx(
+      ['Disabling General Permissions Manager'],
+      async () => {
+        const permissionManager = await st.getPermissionManager();
+        await st.archiveModule(permissionManager.address);
+      },
+      'General Permissions Manager Disabled',
+      () => {
+        dispatch(toggleWhitelistManagement(false));
+        dispatch(loadManagers([]));
+      },
+      undefined,
+      undefined,
+      undefined,
+      true
+    )
+  );
+};
+
+export const addGeneralPermissionModule = () => async (
+  dispatch: Function,
+  getState: GetState
+) => {
+  try {
+    const st: SecurityToken = getState().token.token.contract;
+    const permissionManager = await st.getPermissionManager();
+    const transferManager = await st.getTransferManager();
+    let moduleMetadata = {};
+    let delegateDetails = [];
+
+    if (permissionManager)
+      moduleMetadata = await st.getModule(permissionManager.address);
+
+    dispatch(
+      ui.tx(
+        ['Enabling General Permissions Manager for General Transfer Manager'],
+        async () => {
+          if (moduleMetadata.isArchived) {
+            await st.unarchiveModule(permissionManager.address);
+            delegateDetails = await getDelegateDetails(
+              permissionManager,
+              transferManager
+            );
+          } else {
+            await st.setPermissionManager();
+          }
+        },
+        'General Permissions Manager for General Transfer Manager Enabled',
+        () => {
+          dispatch(loadManagers(delegateDetails));
+          dispatch(toggleWhitelistManagement(true));
+        },
+        undefined,
+        undefined,
+        undefined,
+        true
+      )
+    );
+  } catch (e) {
+    console.log(e);
```

Comment:
Ironically, we might be better off not catching this error, because it will be consumed and lost forever. Perhaps we should let it propagate to the Error Boundary, where it will be sent tp sentry.